### DISCUSSION TITLE
VisibilityConstraints doesn't work properly, or it works just before the command is executed

### DISCUSSION THREAD

**Discussion by aglasencnik** [VsixCommunity/Community.VisualStudio.Toolkit#487] [Q&A] [answered]
Hi, I have a quick question...
I have added the `VisibilityConstraints` to my button, so when it isn't in the right file it is disabled...
```xml
	<VisibilityConstraints>
		<VisibilityItem guid="MyGuid" id="ExportGraphCommand" context="uiContextSupportedFiles"/>
	</VisibilityConstraints>
```
It works perfectly before it is executed for the first time... When I click on the button for the first time, it stays enabled even if I switch to other files, where it shouldn't be enabled.... What should I do?
Is there a way to completely make the button invisible, as this would be ideal scenario?

**Comment by madskristensen** [comment, marked as answer]
Visibility constrains works until your Package is loaded, then the command takes over and you can implement the logic using BeforeQueryStatus, or you can delegate the logic back to the VisibilityConstraint using this method in your `Command`:

```csharp
protected override Task InitializeCompletedAsync()
{
    Command.Supported = false;
    return base.InitializeCompletedAsync();
}
```

**Reply by aglasencnik** [reply]
Great, thank you
